03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,296 @@
+
+```{r}
+# Let's prepare the metadata for the uncollapsed `DESeqData` set object
+metadata_uncollapsed <- metadata %>%
+  dplyr::select(-c("refinebio_title_collapsed", "refinebio_title")) %>%
+  tibble::column_to_rownames("refinebio_accession_code")
+
+# Now we need to collapse the metadata to match the collapsed `DESeqDataSet` object
+metadata_collapsed <- metadata %>%
+  dplyr::select(-c("refinebio_title", "refinebio_accession_code")) %>%
+  dplyr::distinct() %>%
+  # Now assign the collapsed column of identifiers as rownmaes
+  tibble::column_to_rownames("refinebio_title_collapsed")
+```
+
+## 5) Define a minimum counts cutoff
+We are going to do some pre-filtering to keep only rows with 10 or more reads
+total.
+
+```{r Define minimum counts cutoff}
+# First let's take a look at the distribution of counts -- let's use the
+# collapsed replicates data
+collapsed_count_sums <- data.frame(count = rowSums(counts(dds_collapsed)))
+
+ggplot2::ggplot(collapsed_count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+                      color = "blue",
+                      linetype = "dashed") +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count), # This line represents the mean count value
+                                   color = "red",
+                                   linetype = "dashed"))
+
+# Define a minimum counts cutoff -- using the collapsed `DESeqDataSet` object
+genes_to_keep <- rowSums(counts(dds_collapsed)) >= 10
+dds_collapsed <- dds_collapsed[genes_to_keep, ]
+```
+
+```{r}
+# Now let's take a look at the distribution of counts using the uncollapsed
+# replicates data
+uncollapsed_count_sums <- data.frame(count = rowSums(counts(dds)))
+
+ggplot2::ggplot(uncollapsed_count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+                      color = "blue",
+                      linetype = "dashed") +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count), # This line represents the mean count value
+                                   color = "red",
+                                   linetype = "dashed"))
+
+# Define a minimum counts cutoff -- using the uncollapsed `DESeqDataSet` object
+genes_to_keep <- rowSums(counts(dds)) >= 10
+dds <- dds[genes_to_keep, ]
+```
+
+## 6) Perform DESeq2 normalization
+We are going to use a function from the `DESeq2` package to normalize the data. 
+
+```{r Perform DESeq2 normalization}
+# Normalize the data in the `DESeqDataSet` object using the
+# `normTransform` function from the `DESEq2` R package
+## TODO: Add a note/link to section in documentation on why we use
+# `normTransform` vs `vst` and `rlog`
+# Let's normalize the data for the collapsed counts first 
+ntd_collapsed <- normTransform(dds_collapsed)
```

Comment:
I left this TO DO because I am not sure myself. My decision came from the original issue #127 where you mentioned `DESeq2` normalization, so I figured that the `normTransform`, being a function from the `DESeq2` package, was the best to implement here and it was also implemented in the vignette you put on the original issue.

Reply:
When you have time (so maybe not before we get training stuff wrapped up), I would look at these docs for: 
- [`normTransform`](https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/normTransform) 
and see if you can decipher how this is different than 
- [`rlog`](https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/rlog) or 
- [`vst`](https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/varianceStabilizingTransformation)

We can chat about it sometime. The paper might also be helpful to you: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049/

Reply:
Gotcha! :thankyou: